### PR TITLE
Fix/stop process

### PR DESCRIPTION
**Pull request by sifnoc**
Fixing for Issue #114 

Update `circleci` Config and tx test package account.

### CODE REVIEW

**Review by 4000D — COMMENTED** [review]
@sifnoc I think sending `tm.quit` as a parameter is not needed because re-ordering in https://github.com/Onther-Tech/plasma-evm/pull/115/files#diff-d4e7314a5d64e9f6bd6e77e2796e4079L167-R168 would fix it.

**Comment by sifnoc** [comment]
> @sifnoc I think sending `tm.quit` as a parameter is not needed because re-ordering in https://github.com/Onther-Tech/plasma-evm/pull/115/files#diff-d4e7314a5d64e9f6bd6e77e2796e4079L167-R168 would fix it.

I thought that but even if change reordered, still happens(rarely), I will reproduce without checking `tm.quit` channel.

**Comment by sifnoc** [comment]
Reporoduced.
https://app.circleci.com/pipelines/github/sifnoc/plasma-evm/102/workflows/33d194a4-7f14-4263-a6c1-7d813868dddf/jobs/240

Failed at try 3

File: tx/rawdb.go [hunk 1]
```diff
@@ -198,13 +198,18 @@ func ReadNumConfirmedRawTxs(db ethdb.Reader, addr common.Address) uint64 {
 	return n
 }
 
-func WriteNumConfirmedRawTxs(db ethdb.KeyValueWriter, addr common.Address, n uint64) {
+func WriteNumConfirmedRawTxs(db ethdb.KeyValueWriter, quit chan struct{}, addr common.Address, n uint64) {
```
Comment (4000D) [inline tx/rawdb.go]: I think adding `wg` to `rcm` can remove using `quit` channel here, cause db will not closed when `Write*` function is called

Comment (sifnoc) [reply tx/rawdb.go]: Oh my fault. yeah. There is no need checking `quit` chan in `WriteNumConfirmedRawTxs` function. Will fix it.

Comment (sifnoc) [reply tx/rawdb.go]: I forgot to commit this https://github.com/Onther-Tech/plasma-evm/pull/115/commits/a1dab595693bf81601316a6dd8da7d1bd5bf16c4

File: tx/manager.go [hunk 2]
```diff
@@ -256,8 +256,8 @@ func (tm *TransactionManager) Count(account accounts.Account, tx *types.Transact
 	return count
 }
 
-func (tm *TransactionManager) Start() {
-	go tm.confirmLoop()
+func (tm *TransactionManager) Start(wg *sync.WaitGroup) {
```
Comment (4000D) [inline tx/manager.go]: `Start()` func just start's the services, we can make `wg` in `tx.NewManager` func, not from `rcm` .

File: pls/rootchain_manager.go [hunk 3]
```diff
@@ -164,8 +166,9 @@ func (rcm *RootChainManager) Start() error {
 }
 
 func (rcm *RootChainManager) Stop() error {
-	rcm.txManager.Stop()
 	rcm.backend.Close()
+	rcm.txManager.Stop()
+	rcm.wg.Wait()
```
Comment (4000D) [inline pls/rootchain_manager.go]: `rcm` and `rcm.txManager` doesn't need to share `WaitGroup` at all. They can use their own `WaitGroup`. Besides, current `rcm` doesn't need `WaitGroup` so we can make `WaitGroup` only when the `tx.Manager` is created

**Review by 4000D — CHANGES_REQUESTED** [review]

File: pls/rootchain_manager.go (hunk 3 above)
Comment (sifnoc) [reply pls/rootchain_manager.go]: Updated at 656e43d

File: tx/manager.go (hunk 2 above)
Comment (sifnoc) [reply tx/manager.go]: Updated at 656e43d

**Comment by sifnoc** [comment]
Passed Test https://app.circleci.com/pipelines/github/sifnoc/plasma-evm/104/workflows/f18b9431-169b-4d90-9fc8-eb45f36dd763/jobs/246

**Comment by sifnoc** [comment]
Well Tested at https://app.circleci.com/pipelines/github/sifnoc/plasma-evm/107/workflows/236548da-df3e-4cdb-b40d-129dabfba76e/jobs/252

**Pull request merged**